### PR TITLE
Create a sync_nonprofit_users method

### PR DESCRIPTION
**Pull request by camckin10**
At first, and occasionally in the future, we'll want to bulk sync the nonprofit users on the Mailchimp NP list.

Create a "sync_nonprofit_users" method in Mailchimp that queues a ActiveJobs to add each member to the NP list.
Add a spec to show that it works

Closes tix [#4130](https://github.com/CommitChange/tix/issues/4130)

**NOTE: DO NOT discuss internal CommitChange information in your PR; this PR will be public.
Link back to the issue in the Tix repo when you need to do that.**

### CODE REVIEW

File: spec/lib/mailchimp_spec.rb [hunk 1]
```diff
@@ -82,6 +82,14 @@
 		end
   end
 
+	describe 'sync nonprofit users' do 
+		let(:drip_email_list) {create(:drip_email_list)}
+
+		it 'runs nonprofit add user job' do 
+			expect Mailchimp.sync_nonprofit_users(DripEmailList.first)
```
Comment (wwahammy) [inline spec/lib/mailchimp_spec.rb]: I'm not sure I understand what is being checked here.

File: app/legacy_lib/mailchimp.rb [hunk 2]
```diff
@@ -195,6 +195,12 @@ def self.hard_sync_lists(nonprofit, delete_from_mailchimp=false)
     end
   end
 
+  def self.sync_nonprofit_users(drip_email_list)
+    DripEmailList.find_each do |np_user| 
+      MailchimpNonprofitUserAddJob(np_user).perform_now
```
Comment (wwahammy) [inline app/legacy_lib/mailchimp.rb]: This run all of these jobs on the current thread. What we're looking for is to run them all on background threads. We need to queue them up using `#perform_later`

**Review by wwahammy — CHANGES_REQUESTED** [review]
Thanks @camckin10! There are a few changes required here.

File: spec/lib/mailchimp_spec.rb [hunk 3]
```diff
@@ -93,12 +93,6 @@
 			expect(MailchimpNonprofitUserAddJob).to have_been_enqueued.with(drip_email_list, nonprofit_user, nonprofit_user.roles.first.host)
  		end 
 
-		# it 'will NOT include users that doesnt belong to a nonprofit' do 
-		# 	ActiveJob::Base.queue_adapter = :test
-		# 	Mailchimp.sync_nonprofit_users(drip_email_list)
-		# 	expect(MailchimpNonprofitUserAddJob).to_not have_been_enqueued.with(user)
-		# end 
-
```
Comment (wwahammy) [inline spec/lib/mailchimp_spec.rb]: We do need to test this, we didn't have an opportunity to go through the testing however. During our pairing session. I can do work on that today, I think.

**Comment by wwahammy** [comment]
@camckin10 I added back  in the commented out spec and used the [matching arguments](https://rspec.info/features/3-12/rspec-mocks/setting-constraints/matching-arguments/) feature from rspec. Please review the changes let me know if this works for you.

**Comment by camckin10** [comment]
@wwahammy  Looks great! I cannot approve my own PR...So its good to go for me!

**Review by wwahammy — APPROVED** [review]

**Pull request merged**